src/Extension/Console/ExtensionEnableCommand.php

```diff
@@ -90,33 +94,43 @@ protected function allExtensions()
     {
         return $this->extensions->getExtensions()->map(function ($extension) {
             return $extension->getId();
-        });
+        })->toArray();
     }
 
     protected function getExtensionsToEnable()
     {
         $extensionIds = [];
-        if ($this->input->getOption('all')) {
-            // All Extensions
-            $extensionIds = $this->allExtensions();
-        } elseif ($this->input->getOption('include-bundled')) {
-            // Only bundled
-            $extensionIds = $this->allExtensions()->filter(function ($extension) {
-                return substr($extension, 0, 6) === 'flarum' ?: $this->info("Extension: $extension is not bundled, ignoring") && false;
-            });
-        }
 
-        // Process user-specified extensions
         foreach ($this->input->getArgument('extensions') as $extension) {
             if (! $this->extensions->getExtension($extension)) {
-                $this->error("Extension: $extension is not installed.");
-            } elseif (! $extensionIds->contains($extension)) {
+                $this->error("Extension: $extension is not installed, ignoring.");
+            } elseif (! in_array($extension, $extensionIds)) {
                 $extensionIds[] = $extension;
             }
         }
 
-        $extensionIds = $extensionIds->filter(function ($extension) {
-            return ! $this->extensions->isEnabled($extension) ?: $this->info("Extension: $extension is already enabled, ignoring") && false;
+        if ($this->input->getOption('all')) {
+            $extensionIds = $this->allExtensions();
+        } elseif ($this->input->getOption('include-bundled')) {
+            $extensionIds = array_merge($extensionIds, array_filter($this->allExtensions(), function ($extension) use ($extensionIds) {
+                if (substr($extension, 0, 6) !== 'flarum' && ! in_array($extension, $extensionIds)) {
```

Comment:
Same thing here. Too much collision risks with this check IMO

Reply:
Same point as above, but I'd definitely add a "-" after "flarum" in the check